src/core/lib/surface/server.cc

```diff
@@ -1479,6 +1479,10 @@ void Server::ChannelData::InitCall(RefCountedPtr<CallSpineInterface> call) {
           auto* rc = mr.TakeCall();
           rc->Complete(std::move(std::get<0>(r)), *md);
           auto* call_context = GetContext<CallContext>();
+          const auto* deadline = md->get_pointer(GrpcTimeoutMetadata());
+          if (deadline != nullptr) {
+            GetContext<Call>()->UpdateDeadline(*deadline);
```

Comment:
Why do we need to do this both here and in `RecvInitialMetadataReady()` below?  Wouldn't the code there already run before this code does?

Reply:
This is one of the promise based paths (it'll go away very soon too)